workflow/source.go

```diff
@@ -138,6 +148,37 @@ func (d Document) IsRemote() bool {
 	return getFileStatus(d.Location) == fileStatusRemote
 }
 
+func (d Document) IsSpeakeasyRegistry() bool {
+	return strings.Contains(d.Location, "registry.speakeasyapi.dev")
+}
+
+func (d Document) ParseSpeakeasyRegistryReference() *SpeakeasyRegistryDocument {
+	if !d.IsSpeakeasyRegistry() {
+		return nil
+	}
+
+	registryDocument := &SpeakeasyRegistryDocument{}
+	subPath := strings.Split(d.Location, "registry.speakeasyapi.dev/")[1]
+	components := strings.Split(subPath, "/")
+	registryDocument.OrgSlug = components[0]
+	registryDocument.WorkspaceSlug = components[1]
+
+	endComponent := strings.SplitN(components[2], ":", 2)
+	registryDocument.NamespaceName = endComponent[0]
+	reference := endComponent[1]
+	// If the reference is a revision it's expected to be formatted with sha256:...
+	if !strings.HasPrefix(reference, "sha256:") && isLikelyRevision(reference) {
+		reference = "sha256:" + reference
+	}
```

Comment:
a digest reference would be preceded with an at-symbol in the location string to distinguish it from a tag reference. Like so:
```
registry.speakeasyapi.dev/acme/oas/petstore:latest
registry.speakeasyapi.dev/acme/oas/petstore@sha256:902135ed1sa1f2b
```

You would not need `isLikelyRevision` because you'd first try to split on `@` and take everything after that sign as the reference then fallback to splitting on `:` and treating the remainder as a tag reference.

This matches how we refer to images in OCI land.

Reply:
Yup just learned that from Thomas post, will update.